src/query_backend.h

```diff
@@ -1082,11 +1004,26 @@ class Sha1 {
     }
   }
 
-  /// \}
 
   Digest digest() {
     Digest d;
-    hash_.get_digest(d.val);
+    #if BOOST_VERSION_MINOR < 86
+        unsigned int tmp[5];
+    #else
+        unsigned char tmp[20];
+    #endif
+    hash_.get_digest(tmp);
+
+    for (int i = 0; i < CYCLUS_SHA1_NINT; i++) {
+        #if BOOST_VERSION_MINOR < 86
+            d[i] = tmp[i];
+        #else
+            d[i] = (static_cast<uint>(tmp[i*sizeof(unsigned int)]) << 24) |
+                    (static_cast<uint>(tmp[i*sizeof(unsigned int) + 1]) << 16) |
+                    (static_cast<uint>(tmp[i*sizeof(unsigned int) + 2]) << 8) |
+                    (static_cast<uint>(tmp[i*sizeof(unsigned int) + 3]));
```

Comment:
```suggestion
            d[i] = (static_cast<uint>(tmp[i*block_size]) << block_size * 6) |
                   (static_cast<uint>(tmp[i*block_size + 1]) << block_size * 4) |
                   (static_cast<uint>(tmp[i*block_size + 2]) << block_size * 2) |
                   (static_cast<uint>(tmp[i*block_size + 3]));
```

Reply:
Does this work for both big-/little-endian systems?

Reply:
Can we do this as 
```
d[i] = static_cast<uint>(tmp + i * sizeof(unsigned int))
```
And rely on the casting to consume 4 bytes of data?

Reply:
Good idea, this is much cleaner. `static_cast` can do this, and as far as I can tell it should work for both big/little endian systems